aws-customerprofiles-eventstream/src/main/java/software/amazon/customerprofiles/eventstream/UpdateHandler.java

```diff
@@ -0,0 +1,104 @@
+import software.amazon.cloudformation.exceptions.CfnInvalidRequestException;
+import software.amazon.cloudformation.exceptions.CfnNotFoundException;
+import software.amazon.cloudformation.exceptions.CfnServiceInternalErrorException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+@NoArgsConstructor
+public class UpdateHandler extends BaseHandler<CallbackContext> {
+    private CustomerProfilesClient client;
+
+    public UpdateHandler(CustomerProfilesClient client) {
+        this.client = client;
+    }
+
+    @Override
+    public ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+        final AmazonWebServicesClientProxy proxy,
+        final ResourceHandlerRequest<ResourceModel> request,
+        final CallbackContext callbackContext,
+        final Logger logger) {
+        if (this.client == null) {
+            this.client = ClientBuilder.getClient();
+        }
+
+        final ResourceModel requestModel = request.getDesiredResourceState();
+
+        // check whether the event stream exists
+        final GetEventStreamRequest getEventStreamRequest = GetEventStreamRequest.builder()
+            .domainName(requestModel.getDomainName())
+            .eventStreamName(requestModel.getEventStreamName())
+            .build();
+        getEventStream(proxy, logger, requestModel, getEventStreamRequest);
+
+        final Set<Tag> previousTags = request.getPreviousResourceTags() == null ?
+            new HashSet<>() : Translator.mapTagsToSet(request.getPreviousResourceTags());
+
+        if (previousTags != null) {
+            final List<String> tagsToRemove = previousTags.stream()
+                .map(Tag::getKey)
+                .collect(Collectors.toList());
+
+            // remove previous tags
+            if (tagsToRemove.size() > 0) {
+                final UntagResourceRequest untagResourceRequest = UntagResourceRequest.builder()
+                    .resourceArn(Translator.toEventStreamArn(request))
+                    .tagKeys(tagsToRemove)
+                    .build();
+                proxy.injectCredentialsAndInvokeV2(untagResourceRequest, client::untagResource);
+            }
+        }
+
+        if (request.getDesiredResourceTags() != null && !request.getDesiredResourceTags().isEmpty()) {
+            final Map<String, String> resourceTags = request.getDesiredResourceTags();
+            final TagResourceRequest tagResourceRequest = TagResourceRequest.builder()
+                .resourceArn(Translator.toEventStreamArn(request))
+                .tags(resourceTags)
+                .build();
+            proxy.injectCredentialsAndInvokeV2(tagResourceRequest, client::tagResource);
+        }
+
+        return new ReadHandler().handleRequest(proxy, request, callbackContext, logger);
```

Comment:
What's the behaviour of CFN Update for EventStream? Does this indicate we ignore all fields update except for tags?

Reply:
Today we do not have an option update the event stream (no UPDATE API), hence the UpdateHandler only supports updating tags on the existing resource.